File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/AttributeCertificate.java
package aladdin.asn1.iso.pkix;
import aladdin.asn1.*; 
import aladdin.asn1.iso.*; 
import java.io.*;

//	AttributeCertificate ::= SEQUENCE {
//		acinfo               AttributeCertificateInfo,
//		signatureAlgorithm   AlgorithmIdentifier,
//		signatureValue       BIT STRING
//	}

public final class AttributeCertificate extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(AttributeCertificateInfo	.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier		.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(BitString                  .class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public AttributeCertificate(IEncodable encodable) throws IOException { super(encodable, info); } 

	// конструктор при закодировании
	public AttributeCertificate(AttributeCertificateInfo acinfo, 
		AlgorithmIdentifier signatureAlgorithm, BitString signatureValue) 
	{
		super(info, acinfo, signatureAlgorithm, signatureValue); 
	} 
	public final AttributeCertificateInfo	acInfo			  () { return (AttributeCertificateInfo)get(0); } 
	public final AlgorithmIdentifier		signatureAlgorithm() { return (AlgorithmIdentifier	   )get(1); }
	public final BitString                  signatureValue	  () { return (BitString               )get(2); }
}

File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/AttributeCertificate.cs
using System;
using System.Runtime.Serialization;

//	AttributeCertificate ::= SEQUENCE {
//		acinfo               AttributeCertificateInfo,
//		signatureAlgorithm   AlgorithmIdentifier,
//		signatureValue       BIT STRING
//	}

namespace Aladdin.ASN1.ISO.PKIX
{
	[Serializable]
	public class AttributeCertificate : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<AttributeCertificateInfo	>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier		>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<BitString				    >().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected AttributeCertificate(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public AttributeCertificate(IEncodable encodable) : base(encodable, info) {} 

		// конструктор при закодировании
		public AttributeCertificate(AttributeCertificateInfo acinfo, 
			AlgorithmIdentifier signatureAlgorithm, BitString signatureValue) : 
			base(info, acinfo, signatureAlgorithm, signatureValue) {} 

		public AttributeCertificateInfo	ACInfo				{ get { return (AttributeCertificateInfo)this[0]; } } 
		public AlgorithmIdentifier		SignatureAlgorithm  { get { return (AlgorithmIdentifier		)this[1]; } }
		public BitString				SignatureValue		{ get { return (BitString				)this[2]; } }
	}
}
